api/grpc/pb/api.proto

```diff
@@ -59,6 +59,52 @@ message MsgError {
     string error = 1;
 }
 
+message MsgErrorV2 {
+    ErrorCategory category = 1;
+    ErrorCode code = 2;
+    string message = 3;
+    oneof addInfo {
+        ErrV2InfoResourceNotFound ErrV2InfoResourceNotFound = 4;
+    }
+}
+
+enum ErrorCategory {
+    ParticipantError = 0;
+    ClientError = 1;
+    ProtocolError = 2;
+    InternalError = 3;
+}
+
+enum ErrorCode {
+    /* Though "0" is an invalid error code, we still define it, because
+       proto3 requires that every enum definition should have 0 mapped to
+       atleast one constant. */
+    DefaultInvalidCode             = 0;
+
+    ErrV2PeerResponseTimedout      = 101;
```

Comment:
```suggestion
    ErrV2PeerResponseTimedOut      = 101;
```

Reply:
Updated in 47c015884abb63d42121a1c227363cd3db16f387.